crates/re_log_types/src/path/entity_path_filter.rs

```diff
@@ -102,8 +115,8 @@ impl EntityPathFilter {
     /// The rest of the line is trimmed and treated as an entity path.
     ///
     /// Conflicting rules are resolved by the last rule.
-    pub fn parse_forgiving(rules: &str) -> Self {
-        Self::from_query_expressions(rules.split('\n'))
+    pub fn parse_forgiving(rules: &str, subst_env: &HashMap<String, String>) -> Self {
```

Comment:
We pass this `subst_env: &HashMap<String, String>` around in a few places, even outside this file.

It might be worth creating a simple helper type for clarity, e.g.

```rs
#[dervie(Default)]
pub struct EntityPathSubs(pub HashMap<String, String>)

impl EntityPathSubs {
    pub fn new_with_origin(origin: EntityPath) -> Self { … }
}
```